src/lib/y2partitioner/yast_nfs_client.rb

```diff
@@ -0,0 +1,264 @@
+# find current contact information at www.suse.com.
+
+require "yast"
+require "y2storage"
+require "y2partitioner/device_graphs"
+
+Yast.import "Event"
+Yast.import "Stage"
+Yast.import "PackageCallbacks"
+Yast.import "PackageSystem"
+Yast.import "Popup"
+
+module Y2Partitioner
+  # This class handles the interaction between the partitioner and
+  # yast2-nfs-client, making it possible to embed the dialogs from the later into
+  # the corresponding section of the former.
+  class YastNfsClient
+    include Yast::Logger
+    include Yast::I18n
+
+    # Name of the corresponding client in yast2-nfs-client
+    YAST_CLIENT = "nfs-client4part".freeze
+    private_constant :YAST_CLIENT
+
+    # Name of the yast2-nfs-client package
+    PACKAGE = "yast2-nfs-client".freeze
+    private_constant :PACKAGE
+
+    # Whether the client at yast2-nfs-client has read the related system
+    # configuration.
+    #
+    # This is a class method to ensure the configuration is only read once
+    # during the whole partitioner execution.
+    #
+    # @return [Boolean]
+    def self.client_configured?
+      @client_configured
+    end
+
+    # Mark the client configuration as already read.
+    #
+    # @see .client_configured?
+    def self.mark_client_configured
+      @client_configured = true
+    end
+
+    # Resets the information checked by {.client_configured?}
+    #
+    # This is a public class method for testing purposes
+    def self.reset
+      @client_configured = false
+    end
+
+    # Initialize class attributes used by .client_configured?
+    reset
+
+    # Generates user interface from yast2-nfs-client, populated with the NFS
+    # information from the current devicegraph
+    #
+    # If the client is not available, it tries to install it first (with
+    # user confirmation).
+    #
+    # @return [Yast::Term, nil] a term defining the UI, nil if the client is
+    #   not available and was not installed
+    def init_ui
+      return nil unless try_to_ensure_client
+
+      configure_client
+      set_client_list
+      call_client("CreateUI")
+    end
+
+    # Handles an event coming from the UI, consolidating in the current
+    # devicegraph the changes reported by yast2-nfs-client
+    #
+    # @param event [Hash]
+    def handle_input(event)
+      return unless client_exists?
+
+      # It should be already done at this point, but just in case...
+      configure_client
+
+      widget_id = Yast::Event.IsWidgetActivated(event)
+      return if widget_id.nil?
+
+      client_result = call_client("HandleEvent", "widget_id" => widget_id)
+      # FIXME: Note copied from old partitioner (y2-storage) - Take care that
+      # non-fstab settings of nfs-client (firewall, sysconfig, idmapd) get
+      # written on closing partitioner
+      process_result(widget_id, client_result)
+    end
+
+    # Name of the package containing the YaST client
+    #
+    # @return [String]
+    def package_name
+      PACKAGE.dup
```

Comment:
The string is frozen and I would say that's caller responsability to *dup* it if needed. We are creating a new instance everytime we call this method.